### PR TITLE
Tests on Gloss and ProcessedGloss related functions

### PR DESCRIPTION
**Pull request by leandrumartin**
Added tests on the `GrabGlossProperties` and `processGloss` functions in `utils.ts`. These were tested in order to test functionality of the Gloss-related interfaces defined in `Gloss.ts`

### CODE REVIEW

**Comment by leandrumartin** [comment]
I've noticed that the `GrabGlossProperties` function in `utils.ts` takes a Request object. The only two times this function was called (in `useGlossList.ts` and `useGlossInstance.ts`), a request had to be mocked to a relative URL in order to call the function.

I'm confused as to why the function was implemented this way. It looks like all the function does with the request is take the `targetId` from its body. It seemed like it would be more efficient to pass that directly as a parameter instead of mocking a request, so in the latest commit I changed the function to take a string parameter and removed the validation with Zod.

Is there some valid reason why this function should take a Request? Would that make things easier for some other functionality? Because as it stands, it only complicated things in the only two places it was called. If it is better to keep the function as it was, we can simply revert the last commit since that's where I made all the changes to that function.

**Comment by cubap** [comment]
I only started overseeing this code this summer, so I do not know what the original intention was. The changes seem appropriate and it is not too complicated to rebuild something similar if it suddenly becomes important. My best guess is that Tu was working off an example that was built that way and used the mocks to make it work in this case.

File: src/lib/__tests__/gloss.test.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    });
+    const response = await GrabGlossProperties(
+      "https://store.rerum.io/v1/id/614362c3e74876243131a4e0",
+    );
+
+    expect(axios.post).toHaveBeenCalledWith(
+      "https://tinymatt.rerum.io/gloss/query?limit=100&skip=0",
+      {
+        $or: [
+          { target: "http://store.rerum.io/v1/id/614362c3e74876243131a4e0" },
```
Comment (cubap) [inline src/lib/__tests__/gloss.test.js:24]: This is so messy. I think we can schedule some time to correct these targets when we feel like we have it closer to deliverable.

**Review by cubap — APPROVED** [review]
looks good. I think the change makes sense.

**Pull request merged**